### PR TITLE
Update CGW to support Yaml data in nested structure

### PR DESCRIPTION
**Pull request by jalam453**
- Pubtools CGW can support nested structure of data in YAML file
    - Changed the YAML field state to action to denote more appropriate operation
    - File order in YAML will be auto incremented by 10 for nested data structure
    - Updated test cases

### CODE REVIEW

File: pubtools/_content_gateway/utils.py [hunk 1]
```diff
@@ -255,3 +226,56 @@ def sort_items(items):
     sorted_items.extend(product_delete)
 
     return sorted_items
+
+
+def formate_cgw_items(items):
```
Comment (midnightercz) [inline pubtools/_content_gateway/utils.py:231]: formate?
https://en.wikipedia.org/wiki/Formate

**Review by midnightercz — CHANGES_REQUESTED** [review]

File: pubtools/_content_gateway/utils.py (hunk 1 above)
Comment (jalam453) [reply pubtools/_content_gateway/utils.py:231]: It should have been "format". Will update.

File: pubtools/_content_gateway/utils.py [hunk 2]
```diff
@@ -255,3 +226,56 @@ def sort_items(items):
...
+    Args:
+        items (list(dict))
+            list of JSON dictionary
+    Returns:
+        list(dict): list of JSON dictionary
+    """
+
+    formatted_list = list()
+    for product_rec in items:
+        if product_rec.get("type"):
```
[the remark this answers is not in the document]
Comment (jalam453) [reply pubtools/_content_gateway/utils.py:246]: Added comment.

File: docs/entrypoints_reference.rst [hunk 3]
```diff
@@ -118,25 +118,93 @@ A typical YAML format for Product, version and file looks like this:
...
+        productCode: Product_code_1
+        homepage: https://fake.example.com/products/fake-containers/overview/
+        downloadpage: https://fake.example.com/products/fake-containers/download/
+        thankYouPage: https://example.com/
+        thankYouTimeout: 5
+        eloquaCode: fakeCODE9999
+
+        releases:
+          - versionName: 3.4.0
+            action: create
```
[the remark this answers is not in the document]
Comment (jalam453) [reply docs/entrypoints_reference.rst:155]: Updated.

File: docs/entrypoints_reference.rst [hunk 4]
```diff
@@ -118,25 +118,93 @@ A typical YAML format for Product, version and file looks like this:
...
+            masterProductVersion: null
+            ga: true
+            hidden: false
+            invisible: false
+            termsAndConditions: Anonymous Download
+
+            files:
+              - downloadURL: "/content/origin/files/AnsibleNewTest_1/"
+                action: create
```
[the remark this answers is not in the document]
Comment (jalam453) [reply docs/entrypoints_reference.rst:164]: Updated.

File: pubtools/_content_gateway/utils.py [hunk 5]
```diff
@@ -244,17 +244,27 @@ def formate_cgw_items(items):
...
         payload["metadata"] = product_metadata
         product_name = product_metadata["name"]
         product_code = product_metadata["productCode"]
-        formatted_list.append(payload)
+        # nested product data got converted to linear structure
+        formatted_list.append(payload)  # adding record to the result list
 
+        # re-formatting nested versions records in linear order
         for version_rec in product_rec["product"]["releases"]:
             version_payload = {"type": "product_version", "action": version_rec["action"]}
```
Comment (midnightercz) [inline pubtools/_content_gateway/utils.py:269]: action is not available in version_rec, you need to get it from product

**Review by midnightercz — CHANGES_REQUESTED** [review]

File: pubtools/_content_gateway/utils.py (hunk 5 above)
Comment (jalam453) [reply pubtools/_content_gateway/utils.py]: Updated!

**Review by midnightercz — APPROVED** [review]

**Pull request merged**